app/components/view-popup.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+	Dialog,
+	DialogContent,
+	DialogFooter,
+	DialogHeader,
+	DialogTitle,
+	DialogTrigger,
+} from "@/components/ui/dialog";
+import type { DangerKey } from "@/lib/danger-levels";
+import type { Sensor } from "@/lib/sensors";
+import { cn } from "@/lib/utils";
+import { DialogDescription } from "@radix-ui/react-dialog";
+import { t } from "i18next";
+import { PopupNotes } from "./daily-notes";
+import { Button } from "./ui/button";
+import { Card } from "./ui/card";
+import type { Event } from "./weekly-view";
+
+type PopupProps = {
+	title: string;
+	selectedDate: Date | null;
+	exposureData?: PopupData | null;
+	togglePopup?: () => void;
+	handleDayNav?: () => void;
+	children?: React.ReactNode;
+};
+
+export type PopupData = Record<Sensor, DangerKey>;
+
+export function PopupModal({
+	title,
+	selectedDate,
+	exposureData,
+	togglePopup,
+	handleDayNav,
+	children,
+}: PopupProps) {
+	return (
+		<Dialog open={true} onOpenChange={togglePopup}>
+			<DialogTrigger onClick={togglePopup}></DialogTrigger>
+			<DialogContent className="max-w-md">
+				<DialogHeader>
+					<DialogTitle className="font-bold text-xl">{title}</DialogTitle>
+				</DialogHeader>
+				<DialogDescription className="font-medium text-xl">
+					{t("popup.exposureTitle")}
+				</DialogDescription>
+				{selectedDate !== null ? (
+					<div className="flex flex-col gap-2">
+						{exposureData && (
+							<Card className="p-2 md:p-5">
+								{Object.entries(exposureData).map(([sensor, danger]) => (
+									<div key={sensor} className="flex justify-start gap-2">
+										<span
+											className={cn(
+												"rounded-full text-center font-medium capitalize",
+												`bg-${danger} ${danger === "danger" && "text-secondary"}`,
+												"h-fit w-fit px-2",
+											)}
+										>
+											{sensor}
+										</span>
+										<span className="text-muted-foreground">{"->"}</span>
+										<div className={`text-${danger}`}>
+											{t(`popup.${danger}`)}
```

Comment:
You get the point, I wont comment on all of these :)
```suggestion
											{t(($) => $.popup[danger])}
```